src/test/spec/unified_runner/operation.rs

```diff
@@ -1644,6 +1649,142 @@ impl TestOperation for RunCommand {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub(super) struct RunCursorCommand {
+    command: Document,
+    // We don't need to use this field, but it needs to be included during deserialization so that
+    // we can use the deny_unknown_fields tag.
+    #[serde(rename = "commandName")]
+    _command_name: String,
+
+    #[serde(flatten)]
+    options: Option<RunCursorCommandOptions>,
+    session: Option<String>,
+}
+
+impl TestOperation for RunCursorCommand {
+    fn execute_entity_operation<'a>(
+        &'a self,
+        id: &'a str,
+        test_runner: &'a TestRunner,
+    ) -> BoxFuture<'a, Result<Option<Entity>>> {
+        async move {
+            let command = self.command.clone();
+            let db = test_runner.get_database(id).await;
+
+            let max_time = self.options.as_ref().and_then(|opts| opts.max_time);
+            let cursor_type = self.options.as_ref().and_then(|opts| opts.cursor_type);
+            let selection_criteria = self.options.as_ref().and_then(|opts| opts.selection_criteria.clone());
+            let comment = self.options.as_ref().and_then(|opts| opts.comment.clone());
+            let batch_size = dbg!(self.options.as_ref().and_then(|opts| opts.batch_size));
+
+            let options = RunCursorCommandOptions::builder()
+                .max_time(max_time)
+                .cursor_type(cursor_type)
+                .selection_criteria(selection_criteria)
+                .comment(comment)
+                .batch_size(batch_size)
+                .build();
+
+            let result = match &self.session {
+                Some(session_id) => {
+                    with_mut_session!(test_runner, session_id, |session| async {
+                        let mut cursor = db
+                            .run_cursor_command_with_session(command, options, session)
+                            .await?;
+                        cursor.stream(session).try_collect::<Vec<_>>().await
+                    })
+                    .await?
+                }
+                None => {
+                    let cursor = db.run_cursor_command(command, options).await?;
+                    cursor.try_collect::<Vec<_>>().await?
+                }
+            };
+            
+            Ok(Some(bson::to_bson(&result)?.into()))
+        }
+        .boxed()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct CreateCommandCursor {
+    command: Document,
+    // We don't need to use this field, but it needs to be included during deserialization so that
+    // we can use the deny_unknown_fields tag.
+    #[serde(rename = "commandName")]
+    _command_name: String,
+    selection_criteria: Option<SelectionCriteria>,
+    cursor_type: Option<String>,
+    batch_size: Option<u32>,
+    max_time_m_s: Option<u64>,
+    comment: Option<Bson>,
```

Comment:
Once the deserialization issues with `RunCursorCommandOptions` are ironed out can we make the equivalent update here?